python/cugraph/structure/shuffle.py

```diff
@@ -0,0 +1,46 @@
+import math
+from dask.dataframe.shuffle import rearrange_by_column
+
+
+def get_n_workers():
+    from dask.distributed import default_client
+    client = default_client()
+    return len(client.scheduler_info()['workers'])
+
+
+def get_2D_div(ngpus):
+    pcols = int(math.sqrt(ngpus))
+    while ngpus % pcols != 0:
+        pcols = pcols - 1
+    return int(ngpus/pcols), pcols
+
+
+def _set_partitions_pre(df, num_verts, prows, pcols):
+    rows_per_div = math.ceil(num_verts/prows)
+    cols_per_div = math.ceil(num_verts/pcols)
+    partitions = df['src'].floordiv(rows_per_div) * pcols + df['dst'].floordiv(cols_per_div)
+    return partitions
+
+
+def shuffle(dg):
```

Comment:
Suggest adding prows and pcols as parameters to the shuffle function, defaulting to None.

One of the easiest tuning things we want experienced users to be able to do is to configure prows and pcols, and it seems like this would be an easy way to do it.

If prows and pcols are specified then we should use them, otherwise we should compute them.  If a user specifies one then the other should be computable (ngpus / 'whichever one is specified').  If user specifies neither then we can compute as the current code does.

Reply:
added